lib/Sema/ConstraintSystem.cpp

```diff
@@ -2045,6 +2045,9 @@ bool ConstraintSystem::salvage(SmallVectorImpl<Solution> &viable, Expr *expr) {
     SolverState state(expr, *this, FreeTypeVariableBinding::Disallow);
     state.recordFixes = true;
 
+    if (failedConstraint && simplifyConstraint(*failedConstraint) == SolutionKind::Solved)
```

Comment:
The failed constraint is the conforms-to that the CS has identified as the problem, but wasn't able to add a fix for it because `shouldAllowFixes()` has been false up until now. So it definitely needs to be retried.